src/Routing.jl

```diff
@@ -0,0 +1,220 @@
+# general methods for routing rivers
+export compute_streamflow!,
+    update_state_from_hillslope!, update_state_from_channel!
+
+using CSV, DataFrames, Dates, DSP, SpecialFunctions
+
+# methods
+function compute_streamflow!(
+    river_state::RS,
+    river_model::HCM,
+    environment::E,
+    start_date::Date,
+    end_date::Date,
+) where {RS <: RiverState, HCM <: HillslopeChannelRiverModel, E <: Environment}
+
+    update_state_from_hillslope!(
+        river_state,
+        river_model.hillslope_model,
+        environment,
+        start_date,
+        end_date,
+    )
+    update_state_from_channel!(
+        river_state,
+        river_model.channel_model,
+        environment,
+        start_date,
+        end_date,
+    )
+    # return calculate_streamflow(river_state)
```

Comment:
any reason why you moved this into the compare_results.jl function vs kept in here? (the sum of hillslope + channel as the final streamflow)

Reply:
Agreed, in a function called `compute_streamflow` this should return the streamflow as an output.

How about we change this to `update_state!(...)` and then add a function around the outside that does 

```julia
function compute_streamflow(args...)
    update_state!(args)
    return compute_streamflow(river_state) # function from RiverModels.jl
end
```